src/runtime/sgx/common.h

```diff
@@ -0,0 +1,28 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file common.h
+ * \brief TVM SGX common API.
+ */
+#ifndef TVM_RUNTIME_SGX_COMMON_H_
+#define TVM_RUNTIME_SGX_COMMON_H_
+
+#include <tvm/runtime/registry.h>
+#include <string>
+
+namespace tvm {
+namespace runtime {
+namespace sgx {
+
+#define TVM_SGX_CHECKED_CALL(Function)                                         \
+  sgx_status_t TVM_STR_CONCAT(__sgx_status_, __LINE__) = SGX_ERROR_UNEXPECTED; \
```

Comment:
Likely we can remove this file(move macro to sgx_module, do not use PACKED_PREFIX, instead register callback by name+ id that is separate from global function table